numba/targets/npyimpl.py

```diff
@@ -540,3 +541,47 @@ def generate(self, *args):
 
 
 del _kernels
+
+@intrinsic
+def _make_dtype_object(typingctx, desc):
+    """Given a string description *desc*, returns the dtype object.
+    """
+    if isinstance(desc, types.Const):
+        # Convert the str description into np.dtype then to numba type.
+        nb_type = from_dtype(np.dtype(desc.value))
+        return_type = types.DType(nb_type)
+        sig = return_type(desc)
+
+        def codegen(context, builder, signature, args):
+            # All dtype objects are dummy values in LLVM.
+            # They only exist in the type level.
+            return context.get_dummy_value()
+
+        return sig, codegen
+    elif isinstance(desc, types.functions.NumberClass):
+        thestr = str(desc.dtype)
+        # Convert the str description into np.dtype then to numba type.
+        nb_type = from_dtype(np.dtype(thestr))
+        #nb_type = from_dtype(np.dtype(desc.value))
+        return_type = types.DType(nb_type)
+        sig = return_type(desc)
+
+        def codegen(context, builder, signature, args):
+            # All dtype objects are dummy values in LLVM.
+            # They only exist in the type level.
+            return context.get_dummy_value()
+
+        return sig, codegen
+
+@overload(np.dtype)
+def numpy_dtype(desc):
+    if isinstance(desc, types.Const):
```

Comment:
Can these branches be combined like:
```
if isinstance(desc, (types.Const, types.functions.NumberClass)):
```
?

Reply:
Done.